provider/core/src/datagen/iter.rs

```diff
@@ -21,10 +26,15 @@ pub trait IterableDynamicDataProvider<M: DataMarker>: DynamicDataProvider<M> {
 /// return from a load request.
 pub trait IterableDataProvider<M: KeyedDataMarker>: DataProvider<M> {
     /// Returns a list of [`DataLocale`].
-    fn supported_locales(&self) -> Result<Vec<DataLocale>, DataError>;
+    fn supported_requests(&self) -> Result<HashSet<(DataLocale, DataKeyAttributes)>, DataError>;
     /// Returns whether a [`DataLocale`] is in the supported locales list.
-    fn supports_locale(&self, locale: &DataLocale) -> Result<bool, DataError> {
-        self.supported_locales().map(|v| v.contains(locale))
+    fn supports_request(
+        &self,
+        locale: &DataLocale,
+        key_attributes: &DataKeyAttributes,
+    ) -> Result<bool, DataError> {
+        self.supported_requests()
+            .map(|v| v.contains(&(locale.clone(), key_attributes.clone())))
```

Comment:
yes there should be...

Reply:
Could work with `HashMap<DataKeyAttributes, HashSet<DataLocale>>` instead?

Reply:
Or, since you need this tuple everywhere, maybe just export a helper struct called `LocaleWithAttributes` (a standalone helper used in IterableDataProvider but not DataRequest) with all the impls you need.

Reply:
Can I do this in a follow-up? I don't want this to get stale

Reply:
Follow-up is okay

Reply:
#4872